src/_Common/Functions.cs

```diff
@@ -105,6 +105,36 @@ internal static TimeSpan ToTimeSpan(this PeriodSize periodSize)
             };
         }
 
+        internal static PeriodSize? ToPeriodSize(this TimeSpan timeSpan)
+        {
+            PeriodSize? _periodSize = null;
```

Comment:
I've not been using underscores for private variables in our [naming convention](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/naming-guidelines); suggest removing for consistency.